### PR TITLE
Feature/2677 review and map

### PR DESCRIPTION
**Pull request by richard-jones**
Another interim PR for the feature mapping and code review branch.

This one does more than clear out dead code and add documentation. There are a couple of files that have been removed, and their contents factored into other files.  There should be no functional difference, but we should make sure all the tests are passing before merging down.

### CODE REVIEW

**Comment by richard-jones** [comment]
@Steven-Eardley could we put this at quite a high priority to merge, please?  It makes some significant changes to the application form section of the code, so I'd like to get it merged down to develop and ideally then to master to ensure that we do any forward development inline with how the code is now structured.

File: portality/api/v2/discovery.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# ~~APISearch:Feature->API:Feature~~:
```
Comment (Steven-Eardley) [inline portality/api/v2/discovery.py:1]: typo? trailing colon

File: portality/forms/application_processors.py [hunk 2]
```diff
@@ -142,6 +139,29 @@ def _carry_continuations(self):
             # this means that the source doesn't know about current_applications, which is fine
             pass
 
+    def _validate_status_change(self, source_status, target_status):
+        """ Check whether the editorial pipeline permits a change to the target status for a role.
+        Don't run this for admins, since they can change to any role at any time. """
+        from portality.forms.application_forms import application_statuses
+        choices_for_role = [s.get("value") for s in application_statuses(None, self._formulaic)]
+        # choices_for_role = list(sum(application_statuses(None, self._formulaic), ()))
```
Comment (Steven-Eardley) [inline portality/forms/application_processors.py:147]: This file isn't as cleaned up as the others (commented out code)

File: portality/view/forms.py [hunk 3]
```diff
@@ -73,7 +74,6 @@ class EditorGroupForm(Form):
 ## Continuations Forms
 ##########################################################################
 
-ISSN_REGEX = re.compile(r'^\d{4}-\d{3}(\d|X){1}$', re.IGNORECASE)
 ISSN_ERROR = 'An ISSN or EISSN should be 7 or 8 digits long, separated by a dash, e.g. 1234-5678. If it is 7 digits long, it must end with the letter X (e.g. 1234-567X).'
```
Comment (Steven-Eardley) [inline portality/view/forms.py:77]: This should be in messages

**Review by Steven-Eardley — APPROVED** [review]

**Comment by Steven-Eardley** [comment]
This has been merged to develop and is now on the test server, just a small import bug fixed. I haven't functionally tested the application form yet

**Comment by Steven-Eardley** [comment]
Also, the documentation submodule hasn't been pushed; I can supersede that by running the whole suite though

**Pull request merged**